Libraries/libcar/Headers/car/Writer.h

```diff
@@ -35,9 +35,16 @@ class Writer {
     typedef std::unique_ptr<struct bom_context, decltype(&bom_free)> unique_ptr_bom;
 
 private:
+    typedef struct {
+        void *key; size_t key_len; void *value; size_t value_len;
+    } KeyValuePair;
+
+private:
     unique_ptr_bom _bom;
+    ext::optional<struct car_key_format *> _keyfmt;
```

Comment:
Keeping with the style of the existing code. There are no typedefs used in bom_format.h or car_format.h